README.md

```diff
@@ -1,2 +1,112 @@
+[cov-url]: https://coveralls.io/github/ciena-frost/ember-test-utils
+
+[npm-img]: https://img.shields.io/npm/v/ember-test-utils.svg "NPM Version"
+[npm-url]: https://www.npmjs.com/package/ember-test-utils
+
+[![Travis][ci-img]][ci-url] [![Coveralls][cov-img]][cov-url] [![NPM][npm-img]][npm-url]
+
 # ember-test-utils
-A test-framework agnostic set of utilities to aid in testing components
+
+ * [Installation](#Installation)
+ * [Getting Started](#Getting Started)
+ * [Contributing](#Contributing)
+
+## Installation
+```
+ember install ember-test-utils
+```
+
+## Getting Started
+
+**ember-test-utils** provides a set of utilities to help you in testing Ember components.  Wether you're adding integration or unit tests, you can use `registryHelper` to stub out factories.  This library requires you use `ember-cli-mocha` and `ember-mocha` as your testing framework.
+
+### Using `registryHelper`
+
+`registryHelper` must be properly initialized with `beforeSetup` and destroyed with `teardown`
+
+```js
+import {registryHelper} from 'ember-test-utils'
+import resolver from 'path/to/resolver'
+
+describeComponent('component-name', 'description', {
+  unit: true,
+
+  beforeSetup () {
+    registryHelper.setup(resolver, {
+      'component:component-name': ComponentClass
+    })
+  },
+
+  teardown () {
+    registryHelper.teardown()
+  }
+})
+```
+
+`registryHelper.setup` requires some explanation.  It takes a *resolver* as the first argument, since it must restore it when tests are finished.  For a vanilla Ember project, this resolver is usually located at `tests/helpers/resolver`.
+
+The *registry hash* which specifies the factories you want to stub, is an object keyed on the factory name.  The convention for the name is *<type:name>*.
+
+```js
+registryHelper.setup(resolver, {
+  'component:component-name': ComponentClass,
+  'template:components/component-name': hbs`Mr. Fancy Pants`,
+  'helpers:helper-name': HelperFunction,
+  'controller:controller-name': ControllerClass
+})
+```
+
+### Integration vs. Unit tests
+
+`registryHelper` can be used in either integration or unit tests.  Just specify, `unit: true` or `integration: true` in the test module.
+
+If inside an integration tests, all dependencies are loaded/injected.  This environment will allow you to test your components with their dependencies intact.  You can use `registryHelper` to then stub individual dependencies out.
+
+If inside a unit test, the environment is completely isolated.  No dependencies are loaded so you'll need to specify those dependencies with the `needs` array.
+
+
+```js
+describeComponent('component-name', 'description', {
+  unit: true,
+
+  needs: [
+    'component:nested-component-name'
+  ],
+
+  beforeSetup () {
+    registryHelper.setup(resolver, {
+      'component:nested-component-name': NestedComponentClass
+    })
+  },
+
+  teardown () {
+    registryHelper.teardown()
+  }
+})
+```
+
+##Contributing
```

Comment:
Sorry to keep being the Dumb Guy: I think I get what it does and vaguely how it does it. But would love to see an example of what it lets you do? Maybe write an example test, and a short explanation of why that test would have been wicked hard to write or super brittle without the test-utils?